circuits/transfer/src/execute/variants.rs

```diff
@@ -229,218 +277,155 @@ macro_rules! execute_circuit_variant {
+                            dusk_schnorr::gadgets::double_key_verify(
+                                composer,
+                                witness.schnorr_r,
+                                witness.schnorr_r_prime,
+                                witness.schnorr_u,
+                                k,
+                                k_p,
+                                tx_hash,
+                            );
+
+                            // 1.f N == H(k',ip)
+                            let nullifier = sponge::gadget(
+                                composer,
+                                &[*k_p.x(), *k_p.y(), witness.pos],
+                            );
+                            composer.constrain_to_constant(
+                                nullifier,
+                                BlsScalar::zero(),
+                                Some(-input.nullifier()),
+                            );
 
-                    let nullifier_p = sponge::gadget(composer, &[sk_r, pos]);
+                            // 1.g commitment(ic,iv,ib)
+                            let commitment = gadgets::commitment(
+                                composer,
+                                witness.value,
+                                witness.blinding_factor,
+                            );
 
-                    composer.constrain_to_constant(
-                        nullifier_p,
-                        BlsScalar::zero(),
-                        Some(-nullifier),
-                    );
-                });
+                            // 1.h range(iv,64)
+                            composer.range_gate(witness.value, 64);
 
-                // 5. Prove the knowledge of the commitment openings of the
-                // commitments of the input Notes
-                inputs.iter().for_each(|input| {
-                    let value_commitment = input.value_commitment;
-                    let value_commitment_p = gadgets::commitment(
-                        composer,
-                        input.value,
-                        input.blinding_factor,
-                    );
-
-                    composer.assert_equal_point(
-                        value_commitment,
-                        value_commitment_p,
-                    );
-                });
-
-                // 6. Prove that the value of the openings of the commitments of
-                // the input Notes is in range
-                inputs.iter().for_each(|input| {
-                    composer.range_gate(input.value, 64);
-                });
+                            Ok(composer.add(
+                                (BlsScalar::one(), sum),
+                                (BlsScalar::one(), witness.value),
+                                BlsScalar::zero(),
+                                None,
+                            ))
+                        },
+                    )
+                    .map_err(|_| PlonkError::CircuitInputsNotFound)?;
```

Comment:
You don't need `map_err` here, you can just use `or`.

Reply:
Fixed